SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -329,21 +329,40 @@ void USpatialNetDriver::SpatialProcessServerTravel(const FString& URL, bool bAbs
 		}
 	}
 
-	FGuid NextMapGuid = UEngine::GetPackageGuid(FName(*NextMap), GameMode->GetWorld()->IsPlayInEditor());
+	FGuid NextMapGuid = UEngine::GetPackageGuid(FName(*NextMap), World->IsPlayInEditor());
+
+	FString NewURL = URL;
+
+	bool SnapshotOption = NewURL.Contains(TEXT("snapshot="));
+	if (!SnapshotOption)
+	{
+		// In the case that we don't have a snapshot option, we assume the map name will be the snapshot name.
+		// Remove any leading path before the map name.
+		FString Path;
+		FString MapName;
+		NextMap.Split(TEXT("/"), &Path, &MapName, ESearchCase::IgnoreCase, ESearchDir::FromEnd);
+		NewURL.Append(FString::Printf(TEXT("?snapshot=%s"), *MapName));
+	}
 
 	// Notify clients we're switching level and give them time to receive.
-	FString URLMod = URL;
+	FString URLMod = NewURL;
 	APlayerController* LocalPlayer = GameMode->ProcessClientTravel(URLMod, NextMapGuid, bSeamless, bAbsolute);
 
+	// We can't have the NextURL set this early when using SpatialProcessServerTravel so empty the string here.
+	// The reason for this is, on the next WorldTick the current World and NetDriver will be unloaded.
+	// During the deployment wipe we are waiting for an entity query response of all entities in the deployment.
+	// If the NetDriver has been unloaded in that time, the delegate to delete all these entities will be lost and server travel will fail.
+	World->NextURL.Empty();
```

Comment:
This piece of magic fixes a bug related to large world server travelling since the entity query takes longer.